### ISSUE TITLE
'Error ENOENT: module ''dashboard'' reports that it cannot run on the active manager daemon: No module named ''distutils.util'''

### ISSUE THREAD

**Issue by mistur** [ceph/ceph-ansible#5439]
Hello,

What happened:

I made some test with the first 5.0 release of ceph-ansible playbook to deploy an Octopus Cluster, but it failed on "TASK [add modules to ceph-mgr]", is there anything wrong in my playbook setting ?

```
TASK [add modules to ceph-mgr] ***********************************************************************************************************************************************************************************************************************************************************
Monday 22 June 2020  09:11:37 +0200 (0:00:00.083)       0:03:52.591 *********** 
failed: [iccluster024.iccluster.epfl.ch -> iccluster002.iccluster.epfl.ch] (item=dashboard) => changed=true 
  ansible_loop_var: item
  cmd:
  - ceph
  - --cluster
  - lagertha
  - mgr
  - module
  - enable
  - dashboard
  delta: '0:00:00.340429'
  end: '2020-06-22 09:11:38.602366'
  item: dashboard
  msg: non-zero return code
  rc: 2
  start: '2020-06-22 09:11:38.261937'
  stderr: 'Error ENOENT: module ''dashboard'' reports that it cannot run on the active manager daemon: No module named ''distutils.util'' (pass --force to force enablement)'
  stderr_lines: <omitted>
  stdout: ''
  stdout_lines: <omitted>
```

How to reproduce it (minimal and precise):
each time I try to run the playbook with the config bellow

Share your group_vars files, inventory and **full** ceph-ansibe log
[cluster-lagertha.zip](https://github.com/ceph/ceph-ansible/files/4813162/cluster-lagertha.zip)

**Environment**:
* OS (e.g. from /etc/os-release):
```
VERSION="18.04.4 LTS (Bionic Beaver)"
```

* Kernel (e.g. `uname -a`):
```
Linux icitsrv5 4.15.0-55-generic #60-Ubuntu SMP Tue Jul 2 18:22:20 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

* Ansible version (e.g. `ansible-playbook --version`):
```
ansible-playbook 2.9.10.post0
  config file = /home/lagertha/ceph-ansible/ansible.cfg
  configured module search path = ['/home/lagertha/ceph-ansible/library']
  ansible python module location = /home/lagertha/ansible2.9/lib/ansible
  executable location = /home/lagertha/ansible2.9/bin/ansible-playbook
  python version = 3.6.9 (default, Apr 18 2020, 01:56:04) [GCC 8.4.0]
```

* ceph-ansible version (e.g. `git head or tag or stable branch`):
```
stable-5.0
```

* Ceph version (e.g. `ceph -v`):
```
ceph version 15.2.3 (d289bbdec69ed7c1f516e0a093594580a76b78d0) octopus (stable)
```

Thanks for your help.

Best regards,

Yoann

**Comment by dsavineau** [comment]
The `python3-distutils` package should be installed by dependency during the `ceph-mgr-dashboard `package install.

I just get it installed on a bionic node when manually installing the `ceph-mgr-dashboard` deb package.

```console
# dpkg -l python3-distutils
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                                                  Version                         Architecture                    Description
+++-=====================================================-===============================-===============================-================================================================================================================
ii  python3-distutils                                     3.6.9-1~18.04                   all                             distutils package for Python 3.x
```

And I can import the module mentioned in the ansible task failing

```console
# python3 -c 'import distutils.util'
# echo $?
0
```

Could you verify if this package has been installed on yout mgrs nodes ?

**Comment by mistur** [comment]
Hello, 

Here the packages distutils installed I installed them manually, I can try to reinstall all node from scratch. I usually do a minimal install (when I say mininale, it means even vim is not installed) 

```
root@cluster024:~# dpkg -l |  grep distutils
ii  python-distutils-extra                 2.41ubuntu1                                     all          enhancements to the Python build system
ii  python3-d2to1                          0.2.12-1                                        all          Python3 support for distutils2-like setup.cfg files as package metadata
ii  python3-distutils                      3.6.9-1~18.04                                   all          distutils package for Python 3.x
ii  python3-distutils-extra                2.41ubuntu1                                     all          enhancements to the Python3 build system
ii  python3-stsci.distutils                0.3.7-4                                         all          Python3 packaging utilities for STScI's packages
```

looks good to me
```
root@cluster024:~# python3 -c 'import distutils.util'
root@cluster024:~# echo $?
0
```

and the package ceph-mg-dashboard is also installed :
```
root@cluster024:~# dpkg -l | grep  ceph-mgr-dashboard
ii  ceph-mgr-dashboard                     15.2.3-1bionic                                  all          dashboard module for ceph-mgr
```

**Comment by dsavineau** [comment]
> Here the packages distutils installed I installed them manually

I'm not sure if it's coming from your node or from the ansible package installation.

Because installing the `ceph-mgr-dashboard` package pulls automatically the `python3-distutils` package.
But I'm wondering if there's a difference with the ansible package module here (like using the apt no recommends conf, etc...)

**Comment by mistur** [comment]
Hello,

I reinstalled all my node again with a ubuntu 18.04 minimal, then I ran a small script to prepare logical volume and network then I ran the playbook, here the logs : 

[deploy2.log.zip](https://github.com/ceph/ceph-ansible/files/4820422/deploy2.log.zip)

I have exactly the same error.

it looks like python3-distutils has not been installed (I installed it manually in my previous installed after getting the error)

```
lagertha@icitsrv5:~/ceph-ansible$ clush -bw cluster[002,004,006,008,010,012,014,016,018,020,022,024].cluster.local.home "dpkg -l | grep distutils"
clush: cluster[002,004,006,008,010,012,014,016,018,020,022,024].cluster.local.home (12): exited with exit code 1
```

```
lagertha@icitsrv5:~/ceph-ansible$ clush -bw cluster[002,004,006,008,010,012,014,016,018,020,022,024].cluster..local.home "python3 -c 'import distutils.util' 2>&1; echo $?"
---------------
cluster[002,004,006,008,010,012,014,016,018,020,022,024].cluster..local.home (12)
---------------
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'distutils.util'
0
```

Best,

Yoann

**Comment by maltris** [comment]
As of 15.2.5 python3-distutils in bionic is not a direct dependency:

```
Depends: ceph-mgr (= 15.2.5-1bionic), python3-cherrypy3, python3-jwt, python3-bcrypt, python3-werkzeug, python3-routes
```

I had to manually install it afterwards to solve the error. The ceph-mgr-dashboard package should be adjusted.

**Comment by dsavineau** [comment]
@maltris that's not a ceph-ansible issue.

But this has been fixed in the upstream octopus branch [1]

So I guess this will be included in the next 15.2.6 release

[1] https://github.com/ceph/ceph/commit/03cd9e3bb335efc5450cf4a9a3d1db98d54d578b
